CMakeLists.txt

```diff
@@ -2,246 +2,100 @@
 endif( )
 
 # This has to be initialized before the project() command appears
 # Set the default of CMAKE_BUILD_TYPE to be release, unless user specifies with -D.  MSVC_IDE does not use CMAKE_BUILD_TYPE
 if( NOT DEFINED CMAKE_CONFIGURATION_TYPES AND NOT DEFINED CMAKE_BUILD_TYPE )
-  set( CMAKE_BUILD_TYPE "Release" CACHE STRINGS "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel.")
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS None Debug Release RelWithDebInfo MinSizeRel)
+  set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." )
 endif()
 
-# Append our library helper cmake path and the cmake path for hip (for convenience)
-# Users may override HIP path by specifying their own in CMAKE_MODULE_PATH
-list( APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake /opt/rocm/hip/cmake )
+project( rocblas LANGUAGES CXX )
 
-# The superbuild does not build anything itself, all compiling is done in external projects
-project( rocblas-superbuild NONE )
+# This finds the rocm-cmake project, and installs it if not found
+# rocm-cmake contains common cmake code for rocm projects to help setup and install
+set( PROJECT_EXTERN_DIR ${CMAKE_CURRENT_BINARY_DIR}/extern )
+find_package( ROCM CONFIG QUIET PATHS /opt/rocm )
+if( NOT ROCM_FOUND )
+  set( rocm_cmake_tag "master" CACHE STRING "rocm-cmake tag to download" )
+  file( DOWNLOAD https://github.com/RadeonOpenCompute/rocm-cmake/archive/${rocm_cmake_tag}.zip
+    ${PROJECT_EXTERN_DIR}/rocm-cmake-${rocm_cmake_tag}.zip )
 
-# Default ON settings so cmake builds library and samples
-option( BUILD_LIBRARY "Build rocBLAS library" ON )
-option( BUILD_WITH_TENSILE "Building rocBLAS with Tensile or not" ON)
+  execute_process( COMMAND ${CMAKE_COMMAND} -E tar xzf ${PROJECT_EXTERN_DIR}/rocm-cmake-${rocm_cmake_tag}.zip
+    WORKING_DIRECTORY ${PROJECT_EXTERN_DIR} )
 
-option( BUILD_CLIENTS "Build rocBLAS clients" ON )
-option( BUILD_CLIENTS_SAMPLES "Build rocBLAS client samples" ON )
+  find_package( ROCM REQUIRED CONFIG PATHS ${PROJECT_EXTERN_DIR}/rocm-cmake-${rocm_cmake_tag} )
+endif( )
 
-# which benchmark solution schedule
-set( Tensile_ROOT "" CACHE STRING "Local path of Tensile.")
-if ( BUILD_WITH_TENSILE )
-  set( Tensile_LOGIC "FijiROCm15" CACHE STRING "Tensile to use which logic?")
-  set_property( CACHE Tensile_LOGIC PROPERTY STRINGS
-    FijiROCm15
-    )
-  option( Tensile_MERGE_FILES "Tensile to merge kernels and solutions files?" ON)
-  option( Tensile_SHORT_FILENAMES "Tensile to use short file names? Use if compiler complains they're too long." OFF)
-  option( Tensile_PRINT_DEBUG "Tensile to print runtime debug info?" OFF)
-endif()
+include( ROCMSetupVersion )
+include( ROCMCreatePackage )
+include( ROCMInstallTargets )
+include( ROCMPackageConfigHelpers )
+include( ROCMInstallSymlinks )
+
+rocm_setup_version( VERSION 0.5.4.0 )
```

Comment:
Will this need to match the version in the file cmake/build-version.cmake? Does it replace the version in cmake/build-version.cmake?